src/dns.rs

```diff
@@ -208,9 +207,35 @@ cfg_if::cfg_if! {
                         }
                     }
                 }
-            });
 
-            dns_records
+                dns_records
+            }
+
+            match runtime::Handle::try_current() {
+                Ok(handle) => {
+                    // Tokio runtime already exists, cannot block again on the same thread.
+                    // Spawn a new thread to run the blocking code.
+                    let (sender, receiver) = channel();
+                    let name = name.to_owned();
+                    thread::spawn(move || {
+                        let dns_records = handle.block_on(query_with_timeout(name, query_timeout));
+                        // Send the dns_records back to the main thread.
+                        sender.send(dns_records).unwrap();
+                    });
+
+                    // Wait for the dns_records to be sent back from the blocking thread.
+                    receiver.recv().unwrap()
```

Comment:
I would prefer we join the thread and retrieve the result that way instead of using a mpsc channel.
https://doc.rust-lang.org/std/thread/struct.JoinHandle.html#method.join
See also the last example here: https://doc.rust-lang.org/std/thread/fn.spawn.html
What do you think?